tryoutmanager/models.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3.6
+
+import os
+import uuid
+import sqlalchemy as sa
+
+from datetime import datetime
+from sqlalchemy import Column, String, Boolean, DateTime, Table, \
+                       ForeignKey, Enum, Index, UniqueConstraint
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.ext.declarative import declarative_base
+
+db = sa.create_engine(os.environ['DATABASE_URL'])
+metadata = sa.MetaData()
+
+Base = declarative_base(metadata)
+Session = sessionmaker(bind=db)
+
+
+# Association tables for many-to-many relationships
+markers_challenges = Table('markers_challenges',
+                           Base.metadata,
+                           Column('marker_email',
+                                  String,
+                                  ForeignKey('markers.email')),
+                           Column('challenge_id',
+                                  UUID,
+                                  ForeignKey('challenges.id')))
+
+challenges_submissions = Table('challenges_submissions',
+                               Base.metadata,
+                               Column('challenge_id',
+                                      UUID,
+                                      ForeignKey('challenges.id')),
+                               Column('submission_id',
+                                      UUID,
+                                      ForeignKey('submissions.id')))
+
+
+# Tables corresponding to concrete models
+class AdminUser(Base):
+    __tablename__ = 'admin_users'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class Marker(Base):
+    __tablename__ = 'markers'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    challenges = relationship('Challenges',
+                              secondary=markers_challenges,
+                              back_populates='markers')
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class User(Base):
```

Comment:
If we're building for the future here; Users should be able to have public keys.

So we should offer optional github entry so we can automatically fetch them PLUS we'll need to have a local store for them to associate it with a user (IE a table.) Yes, users can have multiple keys.

Reply:
Sounds reasonable. I was actually wondering about how we'd do this (GitHub integration, manual upload, &c.) but wanted to wait to have a conversation about it.

Reply:
So essentially, if they provide us with the username, then we can easily scrap it and place it in their keys. We'd ideally have a `manage my keys` section in a profile management view. 

So I'd say a table that has a key id, owner id, and the key itself?

Reply:
Sounds sane. I'll add a patch that enforces FK between the three user classes (`User`, `Marker`, `AdminUser`) and zero-to-many records from `PublicKey`.

Reply:
👍